openwrt_api.py

```diff
@@ -0,0 +1,400 @@
+  else:
+    raise FileNotFoundError("Could not find /proc/net/dev!")
+
+def get_network_packets(interface):
+  """
+  <Purpose>
+    Count the number of packets received and transmitted from an interface
+
+  <Arguments>
+    interface: the name of the interface to gather network information.
+
+  <Exceptions>
+    RepyArgumentError is raised if the interface aren't valid types or values
+    FileNotFoundError is raised if the file does not exist
+
+  <Side Effects>
+    None
+
+  <Returns>
+    Network usage info as a dict such as {'trans': '598602', 'recv': '1262217'}
+  """
+  if type(interface) is not str:
+    raise RepyArgumentError("get_network_bytes() takes a string as argument.")
+  
+  # Check the input arguments (sanity)
+  if interface not in get_network_interface():
+    raise RepyArgumentError("interface "+ interface + " is not available.")
+
+  if os.path.exists("/proc/net/dev"):
+    dev = safe_open('/proc/net/dev', 'r')
+    for line in dev:
+      if interface in line:
+        data = line.split('%s:' % interface)[1].split()
+        rx_packets, tx_packets = (data[1], data[9])
+        return {"recv": rx_packets, "trans": tx_packets}
+  else:
+    raise FileNotFoundError("Could not find /proc/net/dev!")
+
+def get_network_interface():
+  """
+  <Purpose>
+    Returns a list of available network interfaces.
+
+  <Arguments>
+    None
+
+  <Exceptions>
+    FileNotFoundError is raised if the file does not exist
+
+  <Side Effects>
+    None
+
+  <Returns>
+    The list of strings(interface name).
+  """
+  if os.path.exists("/proc/net/dev"):
+    dev = safe_open("/proc/net/dev", "r").readlines()
+    header_line = dev[1]
+    header_names = header_line[header_line.index("|")+1:].replace("|", " ").split()
+
+    result = []
+    for line in dev[2:]:
+      interface = line[:line.index(":")].strip()
+      result.append(interface)
+    return result
+  else:
+    raise FileNotFoundError("Could not find /proc/net/dev!")
+
+def traceroute(dest_ip,port,max_hops,waittime,ttl):
+  """
+  <Purpose>
+    Print the route packets take to network host 
+
+  <Arguments>
+    dest_ip: The ip address to traceroute
+    port: The port to traceroute
+    max_hops: Maximum number of hops
+    waittime: Set the time to wait for a response to a probe
+    ttl: Set what TTL to start.
+
+  <Exceptions>
+    RepyArgumentError when the arguments aren't valid types or value
+    or values
+
+    ResourceForbiddenError when the local port isn't allowed
+
+
+  <Resource Consumption>
+    This operation consumes 28 bytes + number of bytes of the message that
+    were transmitted. This requires that the localport is allowed.
+
+  <Side Effects>
+    None
+
+  <Returns>
+    The result of traceroute
+  """
+  if type(dest_ip) is not str:
+    raise RepyArgumentError("Provided dest_ip must be a string!")
+
+  if type(port) is not int:
+    raise RepyArgumentError("Provided port must be an integer!")
+
+  if type(max_hops) is not int:
```

Comment:
Also, `max_hops` can't be larger than `255`because the IPv4 header field is 8 bits only.